ui/desktop/src/components/ChatView.tsx

```diff
@@ -94,6 +100,101 @@ export default function ChatView({
     },
   });
 
+  // Listen for make-agent-from-chat event
+  useEffect(() => {
+    const handleMakeAgent = async () => {
+      window.electron.logInfo('Making agent from chat...');
+
+      // Log all messages for now
+      window.electron.logInfo('Current messages:');
+      chat.messages.forEach((message, index) => {
+        const role = isUserMessage(message) ? 'user' : 'assistant';
+        const content = isUserMessage(message) ? message.text : getTextContent(message);
+        window.electron.logInfo(`Message ${index} (${role}): ${content}`);
+      });
+
+      // Inject a question into the chat to generate instructions
+      const instructionsPrompt =
+        'Based on our conversation so far, could you create:\n' +
+        "1. A concise set of instructions (1-2 paragraphs) that describe what you've been helping with. Pay special attention if any output styles or formats are requested (and make it clear), and note any non standard tools used or required.\n" +
+        '2. A list of 3-5 example activities (as a few words each at most) that would be relevant to this topic\n\n' +
+        "Format your response with clear headings for 'Instructions:' and 'Activities:' sections.";
```

Comment:
nit: can probably make this more clear with a small example? should help ensure especially with the smaller models